api/kogitoservices_types.go

```diff
@@ -21,6 +21,46 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 )
 
+// KogitoServiceConditionType is the type of condition
+type KogitoServiceConditionType string
+
+const (
+	// DeployedConditionType - The KogitoService is deployed
+	DeployedConditionType KogitoServiceConditionType = "Deployed"
+	// ProvisioningConditionType - The KogitoService is being provisioned
+	ProvisioningConditionType KogitoServiceConditionType = "Provisioning"
+	// FailedConditionType - The KogitoService is in a failed state
+	FailedConditionType KogitoServiceConditionType = "Failed"
+)
+
+// KogitoServiceConditionReason is the type of reason
+type KogitoServiceConditionReason string
+
+const (
+	// CreateResourceFailedReason - Unable to create the requested resources
+	CreateResourceFailedReason KogitoServiceConditionReason = "CreateResourceFailed"
+	// KogitoInfraNotReadyReason - Unable to deploy Kogito Infra
+	KogitoInfraNotReadyReason KogitoServiceConditionReason = "KogitoInfraNotReadyReason"
+	// ServiceReconciliationFailure - Unable to determine the error
+	ServiceReconciliationFailure KogitoServiceConditionReason = "ReconciliationFailure"
+	// MessagingIntegrationFailureReason ...
+	MessagingIntegrationFailureReason KogitoServiceConditionReason = "MessagingProvisionFailure"
+	// MonitoringIntegrationFailureReason ...
+	MonitoringIntegrationFailureReason KogitoServiceConditionReason = "MonitoringIntegrationFailure"
+	// InternalServiceNotReachable ...
+	InternalServiceNotReachable KogitoServiceConditionReason = "InternalServiceNotReachable"
+	// SuccessfulDeployedReason ...
+	SuccessfulDeployedReason KogitoServiceConditionReason = "AtLeastOnePodAvailable"
+	// FailedDeployedReason ...
+	FailedDeployedReason KogitoServiceConditionReason = "NoPodAvailable"
+	// SuccessfulProvisioningReason ...
+	SuccessfulProvisioningReason KogitoServiceConditionReason = "RequestedReplicasNotEqualToAvailableReplicas"
+	// FailedProvisioningReason ...
+	FailedProvisioningReason KogitoServiceConditionReason = "RequestedReplicasEqualToAvailableReplicas"
```

Comment:
IMHO the reason value should be something more abstract. Provisioning can be set to true also in case there is some  error during reconciliation, while pods are already up and running.
Maybe something like:
```suggestion
	// SuccessfulProvisioningReason ...
	SuccessfulProvisioningReason KogitoServiceConditionReason = "ProvisioningInProgress"
	// FinishedProvisioningReason ...
	FinishedProvisioningReason KogitoServiceConditionReason = "RequestedReplicasEqualToAvailableReplicas"
	// FailedProvisioningReason ...
	FailedProvisioningReason KogitoServiceConditionReason = "ProvisioningFailure"
```